abel/tools/vmi.py

```diff
@@ -243,3 +243,85 @@ def PAD(theta, beta, amplitude):
         amplitude, error_amplitude = np.nan, np.nan
 
     return (beta, error_beta), (amplitude, error_amplitude)
+
+
+def toPES(radial, intensity, energy_cal_factor, per_energy_scaling=True,
+          wavelength=None, Vrep=None, zoom=1):
+    """
+    Convert speed radial coordinate into electron kinetic or binding energy.
+    Return the photoelectron spectrum (PES).
+
+    This calculation uses a single scaling factor ``energy_cal_factor``
+    to convert the radial pixel coordinate into electron kinetic energy.
+
+    Additional experimental parameters, ``wavelength(nm)`` will give the
+    energy scale as the electron binding energy in cm-1, while ``Vrep``,
+    the VMI lens repeller voltage (volts), provides for a voltage
+    independent scaling factor. i.e. ``energy_cal_factor`` should
+    remain approximately constant.
+
+    The ``energy_cal_factor`` is readily determined by comparing the
+    generated energy scale with published spectra. e.g. for O :math:`^-`
+    photodetachment, the strongest fine-structure transition occurs at the
+    electron affinity :math:`EA = 11,784.676(7)` cm :math:`^{-1}`. Values for
+    the ANU experiment are given below, see also
+    `examples/example_hansenlaw.py`.
+
+    Parameters
+    ----------
+    radial : numpy 1D array
+        radial coordinates
+
+    intensity : numpy 1D array
+        intensity values, at the radial array
+
+    energy_cal_factor: float
+        energy calibration factor that will convert radius squared into energy
+        e.g. `1.148427e-5` for "examples/data/O-ANU1024.txt"
+
+    per_energy_scaling : bool
+        sets the intensity Jacobian. If `True`, the returned intensities
+        correspond to an "intensity per eV" or "intensity per cm-1".
+        If `False`, the returned intensities correspond to an "intensity per
+        pixel"
+
+    Optional:
+     wavelength: float
+         measurement wavelength in nm
+         e.g. `812.51 nm`, for "examples/data/O-ANU1024.txt"
+
+     Vrep: float
+         repeller voltage
+         e.g. `-98 volts`, for "examples/data/O-ANU1024.txt"
+
+     zoom: float
+         additional factor if experimental image has been zoomed
+
+
+    """
+
+    if Vrep is not None:
+        energy_cal_factor *= np.abs(Vrep)/zoom**2
+
+    eKE = radial**2*energy_cal_factor
+
+    if wavelength is not None:
+        # electron binding energy
+        eBKE = 1.0e7/wavelength - eKE
+    else:
+        eBKE = eKE
+
+    # Jacobian correction to intensity, radius has been squared
+    # We have E = c1 - c2 * r**2, where c1 and c2 are constants. To get thei
+    # Jacobian, we find dE/dr = 2c2r. Since the coordinates are getting
+    # stretched at high E and "squished" at low E, we know that we need to
+    # divide by this factor.
+    intensity[1:] /= (2*radial[1:])  # 1: to exclude R = 0
```

Comment:
It is great having having your experience, and shared confusion on this ;-)

A very good point about the non-uniform grid. I added a comment to the `Returns` parameter doc.